### PR TITLE
Fix 1946

### LINKED ISSUE #1946: Bugfix: Filter data in `Widget_FlagOverTime.js` instead of storing two dfs
## Expected Behavior

show recent/show all time toggle in the Flag Over Time widget should filter data within the widget to most recent 12 columns (last 12 columns) when show recent is selected

## Current Behavior

we pass in two data frames in `Widget_FlagOverTime.R` to the js widget, which will likely cause performance issues due to large size

## Possible Solution

Just pass in `gtFlagOverTime` and the `most_recent12` array so that we can show either the filtered or the full version of the data depending on the toggle.

## Steps to Reproduce

1.
2.
3.
4.

### LINKED ISSUE #2000: Feature: Add outcome select to `barChart` and `timeSeries` widgets.
## Feature Details
To reduce the size of the KRI report, add a new dropdown to the `barChart` and `timeSeries` widgets that controls which outcome is plotted on the y-axis: `Score`, `Metric`, or `Numerator`.

## Additional Comments
Each widget contains its own input data so 2 bar charts and 3 time series duplicates the data unnecessarily.

### LINKED ISSUE #2001: Feature: Remove `ggplot2` charts from `Visualize_Metric`.
## Feature Details
They're not used, so only adding to overhead.

### PR DESCRIPTION
**Pull request by samussiah**
## Overview
Streamline report to reduce its size with a focus on `Widget_FlagOverTime` and consolidating duplicate `Widget_BarChart` and `Widget_TimeSeries`.

## Test Notes/Sample Code
- `Widget_FlagOverTime`
```
# Include all risk signals, irrespective flag value.
Widget_FlagOverTime(
  dfResults = reportingResults,
  dfMetrics = reportingMetrics
)

# Include risk signals that were ever flagged.
Widget_FlagOverTime(
  dfResults = FilterByFlags(
    reportingResults
  ),
  dfMetrics = reportingMetrics
)

# Include risk signals that were only flagged in the most recent snapshot.
Widget_FlagOverTime(
  dfResults = FilterByFlags(
    reportingResults,
    bCurrentlyFlagged = TRUE
  ),
  dfMetrics = reportingMetrics,
  bExcludeEver = TRUE
)
```
- `Widget_BarChart` / `Widget_TimeSeries`
```
lCharts <- Visualize_Metric(
  dfResults = reportingResults,
  dfBounds = reportingBounds,
  dfGroups = reportingGroups,
  dfMetrics = reportingMetrics,
  strMetricID = "Analysis_kri0001",
  bDebug = TRUE
)
```

## Connected Issues

- Closes #1946 
- Closes #2000 
- Closes #2001

### CODE REVIEW

**Review by lauramaxwell — APPROVED** [review]
This is working well for me! and report size with multiple snapshots is about half the size that dev creates, so definitely an improvement 🎉

**Review by zdz2101 — COMMENTED** [review]
Everything looks much more cleaned up to me, maybe we revisit the toggle (later?) to see most recent N timepoints,, example below would still render a really wide table

```
# Flag Over Time Example
reportingResultsSubset <- dplyr::filter(
  reportingResults,
  GroupID %in% head(unique(reportingResults$GroupID))
)
reportingResultsSubset2 <- reportingResultsSubset %>% 
  mutate(SnapshotDate = SnapshotDate +365)

reportingResultsSubset3 <- dplyr::bind_rows(
  reportingResultsSubset, 
  reportingResultsSubset2
)

Widget_FlagOverTime(
  dfResults = reportingResultsSubset3,
  dfMetrics = reportingMetrics
)
```

**Comment by lauramaxwell** [comment]
> Everything looks much more cleaned up to me, maybe we revisit the toggle (later?) to see most recent N timepoints,, example below would still render a really wide table

good point- @samussiah are we planning on moving the flagovertime widget to a gsm.app plugin where we wouldn't need the toggle, or is it just too much of a pain to implement a filter on the html table in the report?

**Comment by samussiah** [comment]
Updated to limit flag over time widget to the 12 most recent snapshots in the report.

File: inst/report/Report_KRI.Rmd [hunk 1]
```diff
@@ -86,26 +85,36 @@ GroupOverViz <- Widget_GroupOverview(
...
 
 # Add FlagOverTime Table if data has more than one snapshot
-
 if(length(unique((params$dfResults)$SnapshotDate)) > 1){
-  # Remove sites/countries whose metrics have no flags at all timepoints
-  dfResults_flags <- FilterByFlags(params$dfResults)
-  
-  FlagOverViz <- Widget_FlagOverTime(
-    dfResults = dfResults_flags,
+  intSnapshots <- 12L # consider adding this as a parameter
```
Comment (zdz2101) [inline inst/report/Report_KRI.Rmd:93]: Was trying to find the logic for the 12, it was here!

**Review by zdz2101 — APPROVED** [review]

**Pull request merged**